internal/services/appservice/helpers/shared_schema.go

```diff
@@ -1554,7 +1553,6 @@ func StickySettingsSchema() *pluginsdk.Schema {
 					Optional: true,
 					Elem: &pluginsdk.Schema{
 						Type:         pluginsdk.TypeString,
-						ValidateFunc: appserviceValidate.AppSettingName,
```

Comment:
Unfortunately, we can't remove validation entirely, since an empty name is invalid, so we'd need to at least use:
```
							ValidateFunc: validation.StringIsNotEmpty,
```